integreat_cms/core/management/commands/repair_tree.py

```diff
@@ -0,0 +1,184 @@
+import logging
+
+from django.core.management.base import CommandError
+
+from ....cms.models import Page
+from ..log_command import LogCommand
+
+logger = logging.getLogger(__name__)
+
+
+class Command(LogCommand):
+    """
+    Management command to repair a broken Treebeard page tree
+    """
+
+    help = "Search & replace links in the content"
+    pages_seen = None
+
+    def add_arguments(self, parser):
+        """
+        Define the arguments of this command
+
+        :param parser: The argument parser
+        :type parser: ~django.core.management.base.CommandParser
+        """
+        parser.add_argument(
+            "page_id", type=int, help="The ID of a page belonging to the broken tree."
+        )
+        parser.add_argument(
+            "--commit",
+            action="store_true",
+            help="Whether changes should be written to the database",
+        )
+
+    # pylint: disable=arguments-differ
+    def handle(self, *args, page_id, commit, **options):
+        r"""
+        Try to run the command
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param page_id: The page ID (node) of a broken tree
+        :type page_id: int
+
+        :param commit: Whether changes should be written to the database
+        :type commit: bool
+
+        :param \**options: The supplied keyword options
+        :type \**options: dict
+
+        :raises ~django.core.management.base.CommandError: When the input is invalid
+        """
+        try:
+            tree_id = Page.objects.get(id=page_id).tree_id
+        except Page.DoesNotExist as e:
+            raise CommandError(f'The page with id "{page_id}" does not exist.') from e
+        try:
+            root_node = Page.objects.get(tree_id=tree_id, parent=None)
+        except (Page.DoesNotExist, Page.MultipleObjectsReturned) as e:
+            raise CommandError(
+                f'The root page of tree id "{tree_id}" cannot be determined: {e}'
+            ) from e
+        action = "Fixing" if commit else "Detecting problems in"
+        self.print_info(f"{action} tree with id {tree_id}...")
+        self.pages_seen = []
+        self.calculate_left_right_values(root_node, 1, commit)
+        self.check_for_orphans(tree_id)
```

Comment:
```suggestion
            root_node = Page.objects.get(id=page_id)
        except Page.DoesNotExist as e:
            raise CommandError(f'The page with id "{page_id}" does not exist.') from e
        # Traversing to root node
        while root_node.parent:
            root_node = root_node.parent
        action = "Fixing" if commit else "Detecting problems in"
        self.print_info(f"{action} tree with id {root_node.tree_id}...")
        self.pages_seen = []
        self.calculate_left_right_values(root_node, 1, commit)
        self.check_for_orphans(root_node.tree_id)
```
Not sure if that's a bad idea, but we could also traverse the `parent` field to get the root node instead of going via the `tree_id`... this would have the advantage that we really fix the given page, also in the case where the given page is an orphan itself...

Reply:
As we decided that the parent field is our source of truth, this would be the right choice.